### DISCUSSION TITLE
make spreadsheet readonly

### DISCUSSION THREAD

**Discussion by PratyushaTN** [iddan/react-spreadsheet#344] [Ideas]
Hi,

I need to make the spreadsheet readonly for certain scenarios based on a requirement. At the same time, it should also be editable from other views.

Is it possible to make spreadsheet readOnly conditionally?

**Comment by iddan** [comment]
Right now you can make all the cells read only. There is no global flag for it.

**Comment by sazzamac** [comment]
How can all the cells be made read-only?

**Comment by iddan** [comment]
Currently, there is no API for that. A PR to add a global `disabled` prop is welcome.

**Comment by annavandrews** [comment]
I realise this question is very old but in case anyone else looks back at this looking for a work around - I got round this by adding the following to the spreadsheet component: `onKeyDown={e => e.preventDefault()}`

so it would look something like this:
`<Spreadsheet data={data} onKeyDown={e => e.preventDefault()}  />`

This simply stops a user being able to type into the cells.
